lizmap/modules/lizmap/classes/qgisForm.class.php

```diff
@@ -459,6 +459,25 @@ public function setFormDataFromFields($feature)
         return $form;
     }
 
+    /**
+     * Converts the datetime to the format specified in the qgis Project.
+     *
+     * @param string $value The datetime to convert
+     * @param object $edittype The format of the date
+     */
+    public function convertDateTimeToFormat($value, $edittype)
+    {
+        if (!property_exists($edittype, 'options') || !property_exists($edittype->options, 'field_format')) {
+            return $value;
+        }
+        $dateTab = explode('-', $value);
+        $value = $edittype->options->field_format;
+        $value = str_replace('yyyy', $dateTab[0], $value);
+        $value = str_replace('MM', $dateTab[1], $value);
+
+        return str_replace('dd', $dateTab[2], $value);
```

Comment:
```suggestion
        $date = new DateTime($value);
        return $date->format($edittype->options->field_format);
```
I think this is the easiest way to do. https://www.php.net/manual/fr/datetime.construct.php